ag/ag.proto

```diff
@@ -243,15 +243,12 @@ message GradingCriterion {
 message Review {
     uint64 ID = 1;
     uint64 SubmissionID = 2; // foreign key
-    uint64 ReviewerID = 3;   // TODO(meling) is this the UserID of the reviewer? If so, add a doc comment
-    string review = 4; // JSON encoded grading criteria
-    string feedback = 5;
-    bool ready = 6;
-    uint64 score = 7;
-    // TODO(meling) Consider if benchmarks field is needed; it can be obtained from Assignments via submissionID??
-    // TODO(meling) If we need benchmarks here, there must be a foreign key `ReviewID` in `GradingBenchmark`; then we can also remove the gorm tag.
-    repeated GradingBenchmark benchmarks = 8 [(go.field) = {tags: 'gorm:"foreignKey:ID"'}];
-    string edited = 9;
+    uint64 ReviewerID = 3;   // UserID of the reviewer
+    string feedback = 4;
+    bool ready = 5;
+    uint64 score = 6;
+    repeated GradingBenchmark gradingBenchmarks = 7 [(go.field) = {tags: 'gorm:"foreignKey:ReviewID"'}];
```

Comment:
Do we need to specify the tag for foreign key here? If so, we should keep a doc comment explaining why.

Also, I was wondering if we need to repeat this field here since it could also be obtained via the assignments table via submissionID. If this doesn’t work, we should add a comment explaining it. Database ORMs are not easy, so it needs documentation…

Reply:
Benchmarks can probably be obtained via an assignment, the question is how to pass them to the frontend with the corresponding review if a Review type will not have a []GradingBenchmark attribute. 

I guess we can just fetch from the db all the copies of benchmarks for one assignment and all submissions/reviews and add some frontend logic that will select the correct set of benchmarks to display for a single review. But it will result in adding even more complexed frontend logic, is it worth it just to remove a single duplicate field?

Reply:
We can keep duplicate fields. But maybe then we shouldn't keep two copies in the database. That is, an alternative is to have a separate message for the database and another message for passing data to the frontend. Agree that we should keep the frontend logic as simple as possible.

Reply:
Not sure I understand what do you mean by two copies in the database? There is still one `GradingBenchmarks` table in the database as of now. But there will be an initial set of benchmarks created for the assignment, and then a "copy" of the same set for each reviewed submissions. They will of course have different grades and comments for different reviews. 

For example now:
![image](https://user-images.githubusercontent.com/26818902/126498447-2a42901a-6e58-4ba3-8727-ebecf201f3be.png)
the first three belong to the assignment, the next three belong to the review with ID 1. I don't really see the way to avoid having those copies (other then encode them in a single json string as before).

Reply:
ok